lakefile.lean

```diff
@@ -53,39 +53,41 @@ section Scripts
 
 open System
 
-def getCurrentPackage : ScriptM Package := do
-  let ws ← Lake.getWorkspace
-  let [pkg] := ws.packageList.filter (·.dir = ⟨"."⟩) | throw <| IO.userError "Current package not found."
-  return pkg
-
-def importsForLib (lib : Name) : IO String := do
-  let dir ← FilePath.walkDir lib.toString >>= Array.filterM (not <$> ·.isDir)
-  let filePathToImport : FilePath → String := fun fp ↦ fp.toString.takeWhile (· != FilePath.extSeparator) |>.map <|
-    fun c ↦ if c = FilePath.pathSeparator then '.' else c
-  let imports := dir.foldl (init := "") <| fun s f ↦ s ++ s!"import {filePathToImport f}\n"
+def importsForLib (dir : FilePath) (root : Name) : IO String := do
+  let files ← FilePath.walkDir (dir / root.toString) >>= Array.filterM (not <$> ·.isDir)
+  let filePathToImport (fp : FilePath) : String := fp.toString
+    |>.drop dir.toString.length.succ
```

Comment:
Is there a more robust way to achieve this? I would like to drop the prefix corresponding to the directory `dir` and retain the remaining portion of the relative file path.

Reply:
You cloud use what Lake does for globs (i.e., [`forEachModuleIn`](https://github.com/leanprover/lake/blob/e9745480069793e8276158283bddff968426efa3/Lake/Config/Glob.lean#L26-L35)) or just emulate it.

Reply:
Thanks. I have tried a different approach by creating a function `System.FilePath.makeRelativeTo` that performs the necessary check before modifying the path.